### DISCUSSION TITLE
How to set <C-c> to edit the config file instead of closing yazi?

### DISCUSSION THREAD

**Discussion by moshpirit** [sxyazi/yazi#1943] [1. Q&A] [answered]
### What system are you running Yazi on?

Linux X11

### What terminal are you running Yazi in?

kitty 0.37.0

### `yazi --debug` output

```Shell
Yazi
    Version: 0.3.3 (Arch Linux 2024-09-05)
    Debug  : false
    OS     : linux-x86_64 (unix)

Ya
    Version: 0.3.3 (Arch Linux 2024-09-05)

Emulator
    Emulator.via_env: ("xterm-256color", "")
    Emulator.via_csi: Ok(Kitty)
    Emulator.detect : Kitty

Adapter
    Adapter.matches: Kitty

Desktop
    XDG_SESSION_TYPE           : Some("x11")
    WAYLAND_DISPLAY            : None
    DISPLAY                    : Some(":0")
    SWAYSOCK                   : None
    HYPRLAND_INSTANCE_SIGNATURE: None
    WAYFIRE_SOCKET             : None

SSH
    shared.in_ssh_connection: false

WSL
    WSL: false

Variables
    SHELL              : Some("/bin/bash")
    EDITOR             : Some("nvim")
    VISUAL             : None
    YAZI_FILE_ONE      : None
    YAZI_CONFIG_HOME   : None

Text Opener
    default: Some(Opener { run: "$TERMINAL nvim \"$@\" &", block: false, orphan: true, desc: "Open in another terminal", for_: None, spread: true })
    block  : Some(Opener { run: "${EDITOR} \"$@\"", block: true, orphan: false, desc: "Open here", for_: None, spread: true })

Multiplexers
    TMUX               : false
    tmux version       : tmux 3.5a
    ZELLIJ_SESSION_NAME: None
    Zellij version     : 0.41.2

Dependencies
    file             : 5.45
    ueberzugpp       : No such file or directory (os error 2)
    ffmpegthumbnailer: 2.2.3
    magick           : 7.1.1-41
    fzf              : 0.56.3
    fd               : 10.2.0
    rg               : 14.1.1
    chafa            : 1.14.2
    zoxide           : 0.9.6
    7z               : 17.05
    7zz              : No such file or directory (os error 2)
    jq               : 1.7.1


--------------------------------------------------
When reporting a bug, please also upload the `yazi.log` log file - only upload the most recent content by time.
You can find it in the "~/.local/state/yazi" directory.
```

### Describe the question

I'm trying to set it by using `{ on = "<C-c>",    run = "${EDITOR} ${HOME}/.config/yazi/yazi.toml", desc = "Edit Yazi configuration file" },` at `keymap.toml` (in the `[manager]` section), but it doesn't work. I tried executing `nvim .bashrc` within the shell command mode of Yazi, but it doesn't do anything. 

### Validations

- [X] I have searched the existing discussions/issues

**Comment by sxyazi** [comment]
The `shell` command is exactly what you're looking for: https://yazi-rs.github.io/docs/configuration/keymap/#manager.shell

**Reply by moshpirit** [reply]
Thanks for replying!

I'm trying to understand why do I need the `shell` command, since in `keymap.toml` we have lines like `{ run = '${EDITOR} "$@"', desc = "Open here", block = true, for = "unix" },`

Anyway, I'm trying with `{ on = "<C-c>",    run = "shell '${EDITOR} $HOME/.config/yazi/yazi.toml' --confirm", desc = "Edit Yazi configuration file" },`, but I doesn't seem to have any effect. I also tried to directly call the interactive shell on yazi and type `nvim .bashrc`, but it doesn't do anything at all (beyond creating the task).

I tried opening it in a terminal by adding `$TERMINAL` in the beginning of the command, but I found no way to do that in the same window and showing yazi as it was before opening it as `open` does.

 I like this software a lot, I'm trying to understand how it works :)

Again, thank you so much for your time!

**Reply by sxyazi** [reply, marked as answer]
Neovim is an interactive app, so you'll need the `--block` option to make it work
